mozjs/js/rust/magic_codegen/src/lib.rs

```diff
@@ -0,0 +1,144 @@
+#![feature(proc_macro)]
+extern crate proc_macro;
+extern crate syn;
+#[macro_use]
+extern crate quote;
+
+use syn::{ Body, VariantData };
+use proc_macro::TokenStream;
+
+#[proc_macro_derive(MagicDom)]
+pub fn magic_dom(input: TokenStream) -> TokenStream {
+    let s = input.to_string();
+    let ast = syn::parse_macro_input(&s).unwrap();
+
+	let gen = match ast.body {
```

Comment:
Oh, its that you've mixed tabs with spaces. Rust style is to use 4-space indents, not tabs. Let's be consistent with the usual Rust style.